### PR TITLE
Refactor Ethereum store and improve typings

### PR DESCRIPTION
**Pull request by lubej**
Refactor of Ethereum store with improvement of account and network handling. Improves some of the typings as well. 

Based of https://github.com/oasisprotocol/demo-starter/pull/3.

### CODE REVIEW

File: frontend/src/stores/ethereum.ts [hunk 1]
```diff
@@ -91,57 +94,64 @@ export const useEthereumStore = defineStore('ethereum', () => {
...
-  }
+    eth.on('connect', () => (status.value = ConnectionStatus.Connected));
+    eth.on('disconnect', () => (status.value = ConnectionStatus.Disconnected));
+  };
 
-  function checkIsCorrectNetwork() {
-    return window.ethereum.networkVersion.toString() === Network.FromConfig.toString();
+  async function checkIsCorrectNetwork() {
+    return network.value.toString() === Network.FromConfig.toString();
   }
```
Comment (lubej) [inline frontend/src/stores/ethereum.ts:151]: Need to revert this part of code. It should be checking the provider network.

Comment (lubej) [reply frontend/src/stores/ethereum.ts:151]: Actually it's fine, as:
`await eth.connect();`
Is called before, so it should set the current network value.

Comment (lubej) [reply frontend/src/stores/ethereum.ts:151]: As `window.ethereum.networkVersion` is deprecated.

**Review by matevz — APPROVED** [review]
Tested and LGTM

**Pull request merged**